cob_cartesian_controller/src/cartesian_controller_utils.cpp

```diff
@@ -0,0 +1,161 @@
+ ****************************************************************/
+
+#include <cob_cartesian_controller/cartesian_controller_utils.h>
+
+
+geometry_msgs::Pose CartesianControllerUtils::getPose(std::string target_frame, std::string source_frame)
+{
+    geometry_msgs::Pose pose;
+    tf::StampedTransform stamped_transform;
+    
+    stamped_transform = getStampedTransform(target_frame, source_frame);
+    
+    tf::pointTFToMsg(stamped_transform.getOrigin(), pose.position);
+    tf::quaternionTFToMsg(stamped_transform.getRotation(), pose.orientation);
+
+    return pose;
+}
+
+tf::StampedTransform CartesianControllerUtils::getStampedTransform(std::string target_frame, std::string source_frame)
+{
+    tf::StampedTransform stamped_transform;
+    bool transform = false;
+    
+    do
+    {
+        try
+        {
+            ros::Time now = ros::Time::now();
+            tf_listener_.waitForTransform(target_frame, source_frame, now, ros::Duration(0.5));
+            tf_listener_.lookupTransform(target_frame, source_frame, now, stamped_transform);
+            transform = true;
+        }
+        catch (tf::TransformException& ex)
+        {
+            ROS_ERROR("%s",ex.what());
+            ros::Duration(0.1).sleep();
+        }
+    }while(!transform);
+    
+    return stamped_transform;
+}
+
+void CartesianControllerUtils::transformPose(const std::string source_frame, const std::string target_frame, const geometry_msgs::Pose pose_in, geometry_msgs::Pose& pose_out)
+{
+    bool transform = false;
+    geometry_msgs::PoseStamped stamped_in, stamped_out;
+    stamped_in.header.frame_id = source_frame;
+    stamped_in.pose = pose_in;
+    
+    do
+    {
+        try
+        {
+            stamped_in.header.stamp = ros::Time::now();
+
+            tf_listener_.waitForTransform(target_frame, source_frame, stamped_in.header.stamp, ros::Duration(0.5));
+            tf_listener_.transformPose(target_frame, stamped_in, stamped_out);
+            pose_out = stamped_out.pose;
+            transform = true;
+        }
+        catch (tf::TransformException& ex)
+        {
+            ROS_ERROR("%s",ex.what());
+            ros::Duration(0.1).sleep();
+        }
+    }while(!transform);
+}
+
+
+/// Used to check whether the chain_tip_link is close to the target_frame
+/// 'stamped_transform' expreses the transform between the two frames.
+/// Thus inEpsilonArea() returns 'true' in case 'stamped_transform' is "smaller" than 'epsilon'
+// ToDo: Can we simplify this check? 
+//      e.g. use stamped_transform.getOrigin().length() < epsilon
+//      what about orientation distance?
+bool CartesianControllerUtils::inEpsilonArea(tf::StampedTransform stamped_transform, double epsilon)
+{
+    double roll, pitch, yaw;
+    stamped_transform.getBasis().getRPY(roll, pitch, yaw);
+    
+    bool x_okay, y_okay, z_okay = false;
+    bool roll_okay, pitch_okay, yaw_okay = false;
+       
+    x_okay      = std::fabs(stamped_transform.getOrigin().x()) < epsilon;
```

Comment:
definitely...as stated above...eventually I want these check to be replaced by the cob_frame_tracker ActionServer....